app/forms/idv/inherited_proofing/va/form.rb

```diff
@@ -15,32 +15,45 @@ def namespaced_model_name
 
           # Returns the field names based on the validators we've set up.
           def field_names
-            @field_names ||= validators.filter_map do |validator|
-              validator.attributes.first
-            end&.flatten&.uniq&.sort
+            @field_names ||= fields.keys
+          end
+
+          def fields
+            @fields ||= {
+              first_name: { required: true },
+              last_name: { required: true },
+              phone: { required: false },
+              birth_date: { required: true },
+              ssn: { required: true },
+              address_street: { required: true },
+              address_street2: { required: false },
+              address_city: { required: false },
+              address_state: { required: false },
+              address_country: { required: false },
+              address_zip: { required: true },
+            }
+          end
+
+          def required_fields
+            @required_fields ||= fields.filter_map do |field_name, options|
+              field_name if options[:required]
+            end
+          end
+
+          def optional_fields
+            @optional_fields ||= fields.filter_map do |field_name, options|
+              field_name unless options[:required]
+            end
           end
         end
 
-        private_class_method :namespaced_model_name
+        private_class_method :namespaced_model_name, :required_fields, :optional_fields
 
         attr_reader :payload_hash
 
         validate :validate_field_names
 
-        validates :first_name, presence: true, length: { maximum: 255 }
```

Comment:
is it possible to use the `presence: true` alone if that's the motivation behind the changes?

Reply:
Hi @mitchellhenke ty, that was old codez, and a place holder for when I thought validation required more than just :presence. The only thing we're concerned with is the presence of required fields, not the data; the data is being validated by 3rd parties :D. Makes life easy.

Reply:
Is doing `validates :first_name, presence: true` rather than the more metaprogramming-y approach feasible here?

Reply:
Good question, but I believe not. I wanted to avoid having to hard-code field names in case we ever have hash info that uses key names that are not unique across the hash, for example { address1: { street: 'a' }, address2: { street: 'b' }, ...}. We namespace the fields (in the aforementioned example) to :address1_street, address2_street, etc. Each will then have an attr_accessor created for them with no naming conflicts, or headaches for devs having to get creative with naming. Perhaps a little more work, but I imagined it getting lifted at some point to be used elsewhere, if not solely with our IP stuffs.